File: javabackend/src/main/java/mft/dev/javabackend/service/TodoService.java
package mft.dev.javabackend.service;

import mft.dev.javabackend.dto.insert.TodoInsertRequestDTO;
import mft.dev.javabackend.dto.update.TodoUpdateRequestDTO;
import mft.dev.javabackend.model.Todo;

import java.util.List;

public interface TodoService {

    List<Todo> getTodosByUtenteId(long id);

    Todo getTodo(long id);

    Todo insertTodo(long id, TodoInsertRequestDTO dto);

    List<Todo> insertMoltiTodo(long id, List<TodoInsertRequestDTO> dto);

    Todo updateTodo(long id, TodoUpdateRequestDTO dto);

    void deleteTodo(long id);

    void deleteAllTodoByUtenteId(long id);
}

File: kotlinbackend/src/main/kotlin/mft/dev/kotlinbackend/service/TodoService.kt
package mft.dev.kotlinbackend.service

import mft.dev.kotlinbackend.dto.insert.TodoInsertRequestDTO
import mft.dev.kotlinbackend.dto.update.TodoUpdateRequestDTO
import mft.dev.kotlinbackend.model.Todo

interface TodoService {
    fun getTodosByUtenteId(id: Long): MutableList<Todo>

    fun getTodo(id: Long): Todo

    fun insertTodo(id: Long, dto: TodoInsertRequestDTO): Todo

    fun insertMoltiTodo(id: Long, dto: MutableList<TodoInsertRequestDTO>): MutableList<Todo>

    fun updateTodo(id: Long, dto: TodoUpdateRequestDTO): Todo

    fun deleteTodo(id: Long)

    fun deleteAllTodoByUtenteId(id: Long)
}